### PR TITLE
Don't remove the blockchain state in case lastExecutedSeqNum is zero

### PR DESCRIPTION
**Pull request by yuliasherman**
- Avoid a whole state removal after cleaning of metadata
- Don't allow removal of more than one block

### CODE REVIEW

File: kvbc/src/ReplicaImp.cpp [hunk 1]
```diff
@@ -63,19 +63,17 @@ Status ReplicaImp::start() {
...
-                 << ", new m_lastReachableBlock = " << getLastReachableBlockNum());
+      replicaConfig_, m_cmdHandler, m_stateTransfer, m_ptrComm, m_metadataStorage, pm_);
+  const auto lastExecutedSeqNum = m_replicaPtr->getLastExecutedSequenceNum();
+  LOG_INFO(logger, KVLOG(lastExecutedSeqNum));
+  if (!replicaConfig_.isReadOnly && !m_stateTransfer->isCollectingState()) {
+    try {
+      uint64_t removedBlocksNum = replicaStateSync_->execute(logger, *m_kvBlockchain, lastExecutedSeqNum);
+      LOG_INFO(logger, KVLOG(lastExecutedSeqNum, removedBlocksNum, getLastBlockNum(), getLastReachableBlockNum()));
+    } catch (std::exception &e) {
+      std::terminate();
```
Comment (dartdart26) [inline kvbc/src/ReplicaImp.cpp:75]: Does it make sense to log before calling terminate()?

Comment (yuliasherman) [reply kvbc/src/ReplicaImp.cpp:75]: "throw" has an information in it

File: bftengine/src/bftengine/BFTEngine.cpp [hunk 2]
```diff
@@ -116,9 +116,7 @@ IReplica::IReplicaPtr IReplica::createNewReplica(const ReplicaConfig &replicaCon
                                                  IStateTransfer *stateTransfer,
                                                  bft::communication::ICommunication *communication,
                                                  MetadataStorage *metadataStorage,
-                                                 bool &erasedMetadata,
```
Comment (dartdart26) [inline bftengine/src/bftengine/BFTEngine.cpp:119]: Thanks for removing that. It was confusing, especially having the `erasedMetaData` local variable.

**Review by dartdart26 — APPROVED** [review]
Thanks for improving the state sync procedure @yuliasherman !

File: kvbc/src/ReplicaImp.cpp (hunk 1 above)
Comment (dartdart26) [reply kvbc/src/ReplicaImp.cpp:75]: Yes, I see. Will that end up in the log file, though?

Comment (yuliasherman) [reply kvbc/src/ReplicaImp.cpp:75]: I'll add the log just to be sure.

**Pull request merged**